base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -1273,6 +1293,233 @@ where
         Ok(Box::new((tx_id, pre_image, tx_output)))
     }
 
+    /// broadcasts a MinoTari - Tari atomic swap transaction
+    /// # Arguments
+    /// 'dest_pubkey': The Comms pubkey of the recipient node
+    /// 'amount': The amount of MinoTari to send to the recipient
+    /// 'timelock': The time to lock the funds, it should be some value in the future
+    /// 'fee_per_gram': The amount of fee per transaction gram to be included in transaction
+    #[allow(clippy::too_many_lines)]
+    async fn send_blake2_atomic_swap_transaction(
+        &mut self,
+        destination: TariAddress,
+        amount: MicroTari,
+        selection_criteria: UtxoSelectionCriteria,
+        fee_per_gram: MicroTari,
+        message: String,
+        timelock: u64,
+        transaction_broadcast_join_handles: &mut FuturesUnordered<
+            JoinHandle<Result<TxId, TransactionServiceProtocolError<TxId>>>,
+        >,
+    ) -> Result<Box<(TxId, PublicKey, TransactionOutput)>, TransactionServiceError> {
+        let dest_pubkey = destination.public_key();
+        let tx_id = TxId::new_random();
+        // this can be anything, so let's generate a random private key
+        let pre_image = PublicKey::from_secret_key(&PrivateKey::random(&mut OsRng));
+        let hash: [u8; 32] = Blake256::digest(pre_image.as_bytes()).into();
+
+        // let's check that the timelock corresponds to some time in the future
+        let tip_height = self.last_seen_tip_height.unwrap_or(0);
+        if timelock <= tip_height {
+            return Err(TransactionServiceError::Blake2AtomicSwapTransactionError(format!(
+                "Invalid timelock, must be a value in the future"
+            )));
+        }
+
+        // let's create the HTLC script
+        let script = script!(
+            HashBlake256 PushHash(Box::new(hash)) Equal IfThen
+                PushPubKey(Box::new(dest_pubkey.clone()))
+            Else
+                CheckHeightVerify(timelock) PushPubKey(Box::new(self.resources.wallet_identity.node_identity.public_key().clone()))
+            EndIf
+        );
+
+        // Empty covenant
+        let covenant = Covenant::default();
+
+        // Default range proof
+        let minimum_value_promise = MicroTari::zero();
```

Comment:
I would argue that these comments add no value and the comment above `minimum_value_promise` is not accurate
```suggestion
        let covenant = Covenant::default();
        let minimum_value_promise = MicroTari::zero();
```